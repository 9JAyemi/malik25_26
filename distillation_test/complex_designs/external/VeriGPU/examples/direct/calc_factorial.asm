# calculate 5!

li x1 5
li x2 1
li x3 1

loop:
    mul x2, x2, x1
    addi x1, x1, -1
    bne x1, x3, loop

outr x2
halt
