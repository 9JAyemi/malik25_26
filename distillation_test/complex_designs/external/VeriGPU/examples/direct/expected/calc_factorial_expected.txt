out 0 00000000000000000000000001111000 00000078 120
